src/components/studies/session-creator/SessionCreator.tsx

```diff
@@ -262,24 +306,28 @@ const SessionCreator: FunctionComponent<SessionCreatorProps> = ({
               activeSession={getActiveSession(sessions!)}
             ></AssessmentSelector>
           </DialogContent>
-          <DialogActions>
-            <Button onClick={cancelAssessmentSelector}>Cancel</Button>
+          {!isAddingAssessmentToSession && (
+            <DialogActions>
+              <Button onClick={cancelAssessmentSelector}>Cancel</Button>
 
-            <Button
-              variant="contained"
-              onClick={() => {
-                updateAssessments(getActiveSession(sessions)!.id, [
-                  ...getActiveSession(sessions)!.assessments,
-                  ...selectedAssessments,
-                ])
-                setSelectedAssessments([])
-              }}
-            >
-              {!getActiveSession(sessions)
-                ? 'Please select group and session'
-                : `Add  to  ${getActiveSession(sessions)?.name} `}
-            </Button>
-          </DialogActions>
+              <Button
+                variant="contained"
+                onClick={async () => {
+                  setIsAddingAssessmentToSession(true)
+                  await updateAssessments(getActiveSession(sessions)!.id, [
```

Comment:
I think it makes sense to rework this fn call. You are basically creating an array that contains the previous and new assessments and then in your function you are breaking this array up depending on if the assessment is 'previous'.
Why not have a function that takes previous and new assessments. Then you will just need to import the new assessments and then join those 2 arrays